packages/hydrogen/src/storefront.ts

```diff
@@ -324,13 +324,17 @@ export function createStorefrontClient<TI18n extends I18nBase>(
       }),
     };
 
-    const [body, response] = await fetchWithServerCache(url, requestInit, {
+    const promise = fetchWithServerCache(url, requestInit, {
       cacheInstance: mutation ? undefined : cache,
       cache: cacheOptions || CacheShort(),
       shouldCacheResponse: checkGraphQLErrors,
       waitUntil,
     });
 
+    promise.catch(() => {});
+
+    const [body, response] = await promise;
```

Comment:
In theory this should prevent unhandled exceptions. But in my testing it isn't even necessary.